### ISSUE TITLE
block-scoped-var false positive on renaming destructured variables

### ISSUE THREAD

**Issue by vasa-chi** [eslint/eslint#2059]
test.js
```js
(function () {
  "use strict";

  /* global doStuff */
  let {x: y} = {x: 1};
  doStuff(y);
})();
```

Error
```
test.js
  5:10  error  "y" used outside of binding context  block-scoped-var
  6:10  error  "y" used outside of binding context  block-scoped-var

✖ 2 problems (2 errors, 0 warnings)
```

.eslintrc
```json
{
  "env": {
    "es6": true
  },
  "rules": {
    "block-scoped-var": 2
  },
  "ecmaFeatures": {
    "destructuring": true
  }
}
```

```shell
$ eslint --version
v0.17.0
```

**Comment by nzakas** [comment]
Destructuring is hard. :) Thanks for the report, will look into it.

**Comment by vasa-chi** [comment]
Thanks :)

Btw, the same error happens when destructuring in function arguments:

```js
function({x: y}) {
  console.log(y);
}
```

**Comment by xjamundx** [comment]
This gives a similar error:

```js
class Foo {
  *bar() {
  }
}
```

**Comment by nzakas** [comment]
Working on this.

**Comment by vasa-chi** [comment]
Sorry :), but another one:

test.js
```javascript
import {x as y} from "./other.js";

/* global doStuff */
doStuff(y);
```

Error:
```
test.js
  1:8   error  "x" used outside of binding context  block-scoped-var
  1:13  error  "y" used outside of binding context  block-scoped-var
  4:8   error  "y" used outside of binding context  block-scoped-var

✖ 3 problems (3 errors, 0 warnings)
```

.eslintrc
```json
{
  "env": {
    "es6": true
  },
  "rules": {
    "block-scoped-var": 2
  },
  "ecmaFeatures": {
    "destructuring": true,
    "modules": true
  }
}
```

**Comment by vasa-chi** [comment]
So many corner cases...

**Comment by nzakas** [comment]
Please file a separate issue for that (technically not destructuring so not the same issue).
